target/bin/rspamd-dkim

```diff
@@ -0,0 +1,295 @@
+        else
+          _exit_with_error "No selector provided after 'selector' argument"
+        fi
+        ;;
+
+      ( 'domain' )
+        if [[ -n ${2+set} ]]
+        then
+          DOMAIN=${2}
+          _log 'debug' "Domain set to '${DOMAIN}'"
+        else
+          _exit_with_error "No domain provided after 'domain' argument"
+        fi
+        ;;
+
+      ( 'help' )
+        __usage
+        exit 0
+        ;;
+
+      ( '-vv' )
+        # shellcheck disable=SC2034
+        LOG_LEVEL='trace'
+        shift 1
+        _log 'trace' 'Enabled trace-logging'
+        continue
+        ;;
+
+      ( '-v' )
+        # shellcheck disable=SC2034
+        LOG_LEVEL='debug'
+        shift 1
+        _log 'debug' 'Enabled debug-logging'
+        continue
+        ;;
+
+      ( * )
+        __usage
+        _exit_with_error "Unknown option(s) '${1}' ${2:+"and '${2}'"}"
+        ;;
+
+    esac
+
+    shift 2
+  done
+
+  if [[ ${KEYTYPE} == 'ed25519' ]] && [[ ${KEYSIZE} -ne 2048 ]]
+  then
+    _exit_with_error "Chosen keytype ed25519 does not accept the 'keysize' argument"
+  fi
+
+  return 0
+}
+
+function _create_keys
+{
+  # Note: Variables not marked with `local` are used
+  # in other functions (after this function was called).
+  BASE_DIR='/tmp/docker-mailserver/rspamd/dkim'
+
+  if [[ ${KEYTYPE} == 'rsa' ]]
+  then
+    local BASE_FILE_NAME="${BASE_DIR}/${KEYTYPE}-${KEYSIZE}-${SELECTOR}-${DOMAIN}"
+    KEYTYPE_OPTIONS=('-b' "${KEYSIZE}")
+    _log 'info' "Creating DKIM keys of type '${KEYTYPE}' and lenght '${KEYSIZE}' with selector '${SELECTOR}' for domain '${DOMAIN}'"
+  else
+    local BASE_FILE_NAME="${BASE_DIR}/${KEYTYPE}-${SELECTOR}-${DOMAIN}"
+    KEYTYPE_OPTIONS=('-t' "${KEYTYPE}")
+    _log 'info' "Creating DKIM keys of type '${KEYTYPE}' with selector '${SELECTOR}' for domain '${DOMAIN}'"
+  fi
+
+  PUBLIC_KEY_FILE="${BASE_FILE_NAME}.public.txt"
+  PUBLIC_KEY_DNS_FILE="${BASE_FILE_NAME}.public.dns.txt"
+  PRIVATE_KEY_FILE="${BASE_FILE_NAME}.private.txt"
+
+  mkdir -p "${BASE_DIR}"
+  chown _rspamd:_rspamd "${BASE_DIR}"
+
+  # shellcheck disable=SC2310
+  if __do_as_rspamd_user rspamadm \
+      dkim_keygen                 \
+      -s "${SELECTOR}"            \
+      -d "${DOMAIN}"              \
+      "${KEYTYPE_OPTIONS[@]}"     \
+      -k "${PRIVATE_KEY_FILE}"    \
+      >"${PUBLIC_KEY_FILE}"
+  then
+      _log 'info' 'Successfully created DKIM keys'
+      _log 'debug' "Public key written to '${PUBLIC_KEY_FILE}'"
+      _log 'debug' "Private key written to '${PRIVATE_KEY_FILE}'"
+  else
+    _exit_with_error 'Creating keys failed'
+  fi
+}
+
+function _check_permissions
+{
+  # shellcheck disable=SC2310
+  if ! __do_as_rspamd_user ls "${BASE_DIR}" >/dev/null
+  then
+    _log 'warn' "The Rspamd user ('_rspamd') seems to be unable to list files in the keys directory ('${BASE_DIR}') - Rspamd may experience permission errors later"
+  elif ! __do_as_rspamd_user cat "${PRIVATE_KEY_FILE}" >/dev/null
```

Comment:
If the command fails, STDERR should not be shown and redirected to `/dev/null` too?

```suggestion
  elif ! __do_as_rspamd_user cat "${PRIVATE_KEY_FILE}" &>/dev/null
```

Reply:
No, I definitely want the error to be shown! In case of an error, I expect people to copy the log, and having the error shown is very valuable.